File: leetcode-04/solutions/cpp/0003.cpp
class Solution {
 public:
  int lengthOfLongestSubstring(string s) {
    int ans = 0;
    unordered_map<char, int> map;

    int j = 0;
    for (int i = 0; i < s.size(); i++) {
      char c = s[i];
      if (map.count(c)) j = max(j, map[c]);
      ans = max(ans, i - j + 1);
      map[c] = i + 1;
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0003.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = 0
        dict = {}

        j = 0
        for i in range(len(s)):
            c = s[i]
            if c in dict:
                j = max(j, dict[c])
            ans = max(ans, i - j + 1)
            dict[c] = i + 1

        return ans
